src/nfa/noodlify.cc

```diff
@@ -0,0 +1,79 @@
+/* noodlify.cc -- Noodlification of NFAs
+ *
+ * Copyright (c) 2018 Ondrej Lengal <ondra.lengal@gmail.com>
+ *
+ * This file is a part of libmata.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+#include <mata/nfa.hh>
+#include <mata/noodlify.hh>
+
+using namespace Mata::Nfa;
+
+AutSequence SegNfa::noodlify(SegNfa& aut, const Symbol epsilon, bool include_empty)
+{
+    AutSequence noodles{}; // Store the generated noodles.
+
+    // For each depth, get a list of epsilon transitions.
+    Segmentation segmentation{ aut, epsilon };
+    const auto& epsilon_depths{ segmentation.get_epsilon_depths() };
+
+    Nfa noodle{}; // Noodle created for each combination of epsilon transitions.
```

Comment:
What the fuck is noodle?! I would advise to add to the comment some brief explanation what is the noodle, whether it is something well-establish or something own :-)

I know, Lukas spoke about that, but I'm not exactly sure what exactly it is.

Reply:
Good idea. As I am working with them, I did not think about how someone else could have problems deciphering the meaning of noodles at all. I will make sure to add sufficient explanations.

Reply:
Yes, that's called `curse of the knowledge`, you are familiar with the term, since you are working with it for quite some time, but for new users it could be confusing.

Reply:
There is an explanation of what noodle is in the declaration of `noodlify()`. And `noodlify()` then only creates all possible noodles, as written in the documentation for `noodlify()`. I think anyone trying to use `noodlify()` will read the documentation for the function. Afterwards, they will know what a noodle is and what will `noodlify()` do.

I think this explanation is enough to give our users a good enough understanding of the function.

Reply:
Yeah, sorry, didn't notice it in the comment. That was actually what I was expecting :-)